modules/contracts/apps/challenge/src/msg.rs

```diff
@@ -0,0 +1,114 @@
+use crate::{
+    contract::ChallengeApp,
+    state::{ChallengeEntry, CheckIn, Friend, Vote},
+};
+use abstract_core::objects::AssetEntry;
+use cosmwasm_schema::QueryResponses;
+use cosmwasm_std::Uint128;
+use croncat_app::croncat_integration_utils::CronCatInterval;
+
+abstract_app::app_msg_types!(ChallengeApp, ChallengeExecuteMsg, ChallengeQueryMsg);
+
+#[cosmwasm_schema::cw_serde]
+pub enum Frequency {
+    Daily,
+    Weekly,
+    Monthly,
+    EveryNBlocks(u64),
+}
+
+impl Frequency {
+    pub fn to_interval(self) -> CronCatInterval {
+        match self {
+            Frequency::EveryNBlocks(blocks) => CronCatInterval::Block(blocks),
+            Frequency::Daily => unimplemented!(),
+            Frequency::Weekly => unimplemented!(),
+            Frequency::Monthly => unimplemented!(),
+        }
+    }
+}
+
+/// App instantiate message
+#[cosmwasm_schema::cw_serde]
+pub struct AppInstantiateMsg {
+    /// Native gas/stake asset for this chain
+    pub native_asset: AssetEntry,
+    /// Amount in native coins to forfeit when a challenge is lost
+    pub forfeit_amount: Uint128,
+}
+
+/// App execute messages
+#[cosmwasm_schema::cw_serde]
+#[cfg_attr(feature = "interface", derive(cw_orch::ExecuteFns))]
+#[cfg_attr(feature = "interface", impl_into(ExecuteMsg))]
+pub enum ChallengeExecuteMsg {
+    CreateChallenge {
+        challenge: ChallengeEntry,
+    },
+    UpdateChallenge {
+        challenge_id: String,
+        challenge: ChallengeEntry,
+    },
+    CancelChallenge {
+        challenge_id: String,
+    },
+    AddFriendForChallenge {
```

Comment:
This variant is redundant IMO as one can call `AddFriendsForChallenge` just the same. Same as `RemoveFriendForChallenge`. I'd prefer to see a `RemoveFriendsForChallenge` variant or you could merge those two and have an `UpdateFriendsForChallenge` which takes in friends to add and friends to remove. In general minimizing the API to the functional minimum is the approach we've been trying to stick to.